conans/test/functional/toolchain/test_cmake.py

```diff
@@ -0,0 +1,325 @@
+class WinTest(Base):
+    @parameterized.expand([("Debug", "MTd", "15", "14", "x86", "v140", True),
+                           ("Release", "MD", "15", "17", "x86_64", "", False)])
+    def test_toolchain_win(self, build_type, runtime, version, cppstd, arch, toolset, shared):
+        settings = {"compiler": "Visual Studio",
+                    "compiler.version": version,
+                    "compiler.toolset": toolset,
+                    "compiler.runtime": runtime,
+                    "compiler.cppstd": cppstd,
+                    "arch": arch,
+                    "build_type": build_type,
+                    }
+        options = {"shared": shared}
+        install_out = self._run_build(settings, options)
+        self.assertIn("ERROR: fPIC option defined for Windows. Remove it.", install_out)
+
+        # FIXME: Hardcoded VS version and partial toolset check
+        self.assertIn('CMake command: cmake -G "Visual Studio 15 2017" '
+                      '-DCMAKE_TOOLCHAIN_FILE="conan_toolchain.cmake"', self.client.out)
+        if toolset == "v140":
+            self.assertIn("Microsoft Visual Studio 14.0", self.client.out)
+        else:
+            self.assertIn("Microsoft Visual Studio/2017", self.client.out)
+        if shared:
+            self.assertIn("app_lib.dll", self.client.out)
+        else:
+            self.assertNotIn("app_lib.dll", self.client.out)
+
+        out = str(self.client.out).splitlines()
+        runtime = "MT" if "MT" in runtime else "MD"
+        generator_platform = "x64" if arch == "x86_64" else "Win32"
+        arch = "x64" if arch == "x86_64" else "X86"
+        shared_str = "ON" if shared else "OFF"
+        vals = {"CMAKE_GENERATOR_PLATFORM": generator_platform,
+                "CMAKE_BUILD_TYPE": "",
+                "CMAKE_CXX_FLAGS": "/MP1 /DWIN32 /D_WINDOWS /W3 /GR /EHsc",
+                "CMAKE_CXX_FLAGS_DEBUG": "/%sd /Zi /Ob0 /Od /RTC1" % runtime,
+                "CMAKE_CXX_FLAGS_RELEASE": "/%s /O2 /Ob2 /DNDEBUG" % runtime,
+                "CMAKE_C_FLAGS": "/MP1 /DWIN32 /D_WINDOWS /W3",
+                "CMAKE_C_FLAGS_DEBUG": "/%sd /Zi /Ob0 /Od /RTC1" % runtime,
+                "CMAKE_C_FLAGS_RELEASE": "/%s /O2 /Ob2 /DNDEBUG" % runtime,
+                "CMAKE_SHARED_LINKER_FLAGS": "/machine:%s" % arch,
+                "CMAKE_EXE_LINKER_FLAGS": "/machine:%s" % arch,
+                "CMAKE_CXX_STANDARD": cppstd,
+                "CMAKE_CXX_EXTENSIONS": "OFF",
+                "BUILD_SHARED_LIBS": shared_str}
+        for k, v in vals.items():
+            self.assertIn(">> %s: %s" % (k, v), out)
+
+        toolchain = self.client.load("build/conan_toolchain.cmake")
+        include = self.client.load("build/conan_project_include.cmake")
+        settings["build_type"] = "Release" if build_type == "Debug" else "Debug"
+        self._run_build(settings, options)
+        # The generated toolchain files must be identical
```

Comment:
```suggestion
        # The generated toolchain files must be identical because it is a multi-config
        # generator and information about the build_type is only available to the 'build' stage.
```